pkg/controller/render/render_controller.go

```diff
@@ -524,9 +524,9 @@ func (ctrl *Controller) syncGeneratedMachineConfig(pool *mcfgv1.MachineConfigPoo
 
 // generateRenderedMachineConfig takes all MCs for a given pool and returns a single rendered MC. For ex master-XXXX or worker-XXXX
 func generateRenderedMachineConfig(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.MachineConfig, cconfig *mcfgv1.ControllerConfig) (*mcfgv1.MachineConfig, error) {
-	// Before merging all MCs for a specific pool, let's make sure each contains a valid Ignition Config
+	// Before merging all MCs for a specific pool, let's make sure MachineConfigs are valid
 	for _, config := range configs {
-		if err := ctrlcommon.ValidateIgnition(config.Spec.Config); err != nil {
+		if err := ctrlcommon.ValidateMachineConfig(config.Spec); err != nil {
```

Comment:
Q: if I have a config with an unsupported kernel type will this fail rendering and I need to fix it?  i thought it was not supposed to fail rendering but instead just use the default and let the user know that we are using default kernel instead of the one they have?

Reply:
Failing rendering was what I suggested - at least until we get to the point of having API validation and rejecting at submission time.

Basically we only support the three values "", "default", and "realtime" - we shouldn't silently ignore a typo for "realtiime" e.g.

This is all the same as how we're validating basic Ignition before trying to roll it out, right/

Reply:
sorry this pr is confusing to me, so it _will_ fail if we have an unsupported kernel type not just use the default? (that canonicalizeKernelType function confused me)

Reply:
so canonicalizeKernelType is really covering the "" case and ensuring that default is used for that?

Reply:
yes, canonicalizeKernelType is for code simplification and just for converting "" to default so that we don't un-necessary keep doing comparison for both "" and default.